go/logic/orchestrator.go

```diff
@@ -442,14 +442,23 @@ func SubmitMastersToKvStores(clusterName string, force bool) (kvPairs [](*kv.KVP
 		submitKvPairs = append(submitKvPairs, kvPair)
 	}
 	log.Debugf("kv.SubmitMastersToKvStores: submitKvPairs: %+v", len(submitKvPairs))
+	var kvPutKVPairs [](*kv.KVPair)
 	for _, kvPair := range submitKvPairs {
 		if orcraft.IsRaftEnabled() {
 			_, err = orcraft.PublishCommand("put-key-value", kvPair)
+			if err == nil {
+				submittedCount++
+			} else {
+				selectedError = err
+			}
 		} else {
-			err = kv.PutKVPair(kvPair)
+			kvPutKVPairs = append(kvPutKVPairs, kvPair)
 		}
+	}
+	if len(kvPutKVPairs) > 0 {
+		err := kv.PutKVPairs(kvPutKVPairs)
 		if err == nil {
-			submittedCount++
+			submittedCount = submittedCount + len(kvPutKVPairs)
```

Comment:
I'm not sure I follow the logic here: iterate kvpairs, and if we're on in raft-mode, then it's a no-op, we generate and append to a new slice, which is then tested and applied?

How about this alternative pseudo code:
```
if orcraft.IsRaftEnabled() {
  for _, kvPair := range submitKvPairs {
    _, err = orcraft.PublishCommand("put-key-value", kvPair)
    ...
  }
} else {
  err := kv.submitKvPairs(kvPutKVPairs)
  ...
}
```
this way we don't need the extra slice `kvPutKVPairs` and the logic is simplified. What do you think?

Reply:
@shlomi-noach thanks for reviewing

Yes, on second look this makes no sense 😄 - for some reason I thought I needed to filter each KV against `orcraft.IsRaftEnabled()` but in hindsight that setting is all or nothing

I'll update with the simpler pseudo code you proposed 👍